ImGuiFileDialog.cpp

```diff
@@ -1,72 +1,100 @@
-#include "ImGuiFileDialog.h"
+// From https://github.com/aiekick/ImGuiFileDialog
+//
+// Modified by AlastairGrowcott@yahoo.com to work on Posix systems.
+
+
+#include <sys/types.h>
+#include <dirent.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <algorithm>
+
+#include "imgui_file_dlg.h"
 #include "imgui.h"
-#include "dirent/dirent.h"
 
 #define IMGUI_DEFINE_MATH_OPERATORS
 #include "imgui_internal.h"
 
-#include <algorithm>
 
-inline bool ReplaceString(std::string& str, const std::string& oldStr, const std::string& newStr)
+#define PATH_SEP '/'
+
+
+static const std::string s_fs_root(1u, PATH_SEP);
+
+
+inline void ReplaceString(std::string& str, const std::string& oldStr, const std::string& newStr)
 {
-	bool Finded = false;
-	size_t pos = 0;
-	while ((pos = str.find(oldStr, pos)) != std::string::npos) {
-		Finded = true;
-		str.replace(pos, oldStr.length(), newStr);
-		pos += newStr.length();
-	}
-	return Finded;
+    size_t pos = 0;
+
+    while ((pos = str.find(oldStr, pos)) != std::string::npos) {
+        str.replace(pos, oldStr.length(), newStr);
+        pos += newStr.length();
+    }
 }
 
 inline std::vector<std::string> splitStringVector(const std::string& text, char delimiter)
 {
-	std::vector<std::string> arr;
-	std::string::size_type start = 0;
-	std::string::size_type end = text.find(delimiter, start);
-	while (end != std::string::npos)
-	{
-		std::string token = text.substr(start, end - start);
-		if (token != "")
-			arr.push_back(token);
-		start = end + 1;
-		end = text.find(delimiter, start);
-	}
-	arr.push_back(text.substr(start));
-	return arr;
+    std::vector<std::string>  arr;
+    size_t                    start = 0u;
+    size_t                    end   = text.find(delimiter, start);
+
+    while (end != std::string::npos)
+    {
+        std::string token("");
+
+        if (0u == end)
+        {
+            token += delimiter;
+        }
+        else
+        {
+            token = text.substr(start, end - start);
+        }
+
+        if (!token.empty())
+        {
+            arr.push_back(token);
+        }
+
+        start = end + 1;
+        end = text.find(delimiter, start);
+    }
+    arr.push_back(text.substr(start));
+    return arr;
 }
 
-inline void AppendToBuffer(char* vBuffer, int vBufferLen, std::string vStr)
+inline void AppendToBuffer(char* vBuffer, size_t vBufferLen, std::string vStr)
```

Comment:
Try to always use size_t for sizes. Microsoft loves its integers for everything even when the value will NEVER be negative.